documentation/Architecture/The-Data-Processing-Pipeline.md

```diff
@@ -1,43 +1,131 @@
 # The Data Processing Pipeline
 
-The Custom Data Source (CDS) uses a Data Processor to read from the Data Source. Since the plugins are dynamically loaded into the Driver by the SDK,
-we want to ensure that the Data Processors are efficient in reading and formatting the raw data.
+Within a plugin, a `CustomDataSource` delegates the task of processing data sources to a `CustomDataProcessor`. 
+Because data sources are typically files on disk that need to be both read and parsed, we want to ensure that these 
```

Comment:
"Since data sources are usually files on disk that need to be read and parsed, we want to ensure that these
processors are efficient by accessing the physical disk as infrequently as possible" -- makes the sentence more concise (also don't use because at the beginning of a sentence).

Reply:
Is there something grammatically incorrect about starting a sentence with "because?" I've never heard of that rule 😨